encord/objects/bitmask.py

```diff
@@ -0,0 +1,176 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, List, Sequence
+
+from encord.exceptions import EncordException
+
+
+def _string_to_rle(s: str) -> List[int]:
+    """
+    COCO-compatible string to RLE-encoded mask de-serialisation
+    """
+    cnts = []
+    p = 0
+
+    while p < len(s):
+        x = 0
+        k = 0
+        more = 1
+
+        while more and p < len(s):
+            c = ord(s[p]) - 48
+            x |= (c & 0x1F) << (5 * k)
+            more = c & 0x20
+            p += 1
+            k += 1
+
+            if not more and (c & 0x10):
+                x |= -1 << (5 * k)
+
+        if len(cnts) > 2:
+            x += cnts[-2]
+
+        cnts.append(x)
+
+    return cnts
+
+
+def _rle_to_string(rle: Sequence[int]) -> str:
+    """
+    COCO-compatible RLE-encoded mask to string serialisation
+    """
+    rle_string = ""
+    for i, x in enumerate(rle):
+        if i > 2:
+            x -= rle[i - 2]
+
+        more = 1
+        while more:
+            c = x & 0x1F
+            x >>= 5
+
+            if c & 0x10:
+                more = x != -1
+            else:
+                more = x != 0
+
+            if more:
+                c |= 0x20
+
+            c += 48
+            rle_string += chr(c)
+
+    return rle_string
+
+
+def _rle_to_mask(rle: List[int], size: int) -> bytes:
+    """
+    COCO-compatible RLE to bitmask
+    """
+    res = bytearray(size)
+    offset = 0
+
+    for i, c in enumerate(rle):
+        v = i % 2
+        while c > 0:
+            res[offset] = v
+            offset += 1
+            c -= 1
+
+    return bytes(res)
+
+
+def _mask_to_rle(mask: bytes) -> List[int]:
+    """
+    COCO-compatible raw bitmask to RLE
```

Comment:
I'm wondering if this is the more accurate description
```suggestion
    COCO-compatible raw bitmask to COCO-compatible RLE
```